### PR TITLE
DynamoDBLogStore Java Refactor setup

### PR DESCRIPTION
**Pull request by scottsand-db**
- create dynamodbLogStore SBT module
- create basic skeleton for java implementation

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -173,6 +175,19 @@ lazy val storage = (project in file("storage"))
...
+// TODO(scott): figure out a better way to include tests in this project
+lazy val storageDynamodb = (project in file("storage-dynamodb"))
+  .dependsOn(storage)
+  .settings (
+    name := "delta-storage-dynamodb",
+    commonSettings,
+    skipReleaseSettings,
+    // releaseSettings,
+    libraryDependencies ++= Seq(
+      "com.amazonaws" % "aws-java-sdk" % "1.7.4"
```
Comment (tdas) [inline build.sbt:187]: what made you decide 1.7.4? i am curious.

File: build.sbt [hunk 2]
```diff
@@ -173,6 +175,19 @@ lazy val storage = (project in file("storage"))
     )
   )
 
+// TODO(scott): figure out a better way to include tests in this project
+lazy val storageDynamodb = (project in file("storage-dynamodb"))
+  .dependsOn(storage)
+  .settings (
+    name := "delta-storage-dynamodb",
+    commonSettings,
+    skipReleaseSettings,
```
Comment (tdas) [inline build.sbt:184]: why skip release?

File: core/src/test/scala/org/apache/spark/sql/delta/ExternalLogStoreSuite.scala [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+// TODO: make non-abstract and implement when tests when DynamoDBLogStore is ready
+abstract class ExternalLogStoreSuite extends PublicLogStoreSuite {
```
Comment (tdas) [inline core/src/test/scala/org/apache/spark/sql/delta/ExternalLogStoreSuite.scala:20]: why is this ExternalLogStoreSuite and not DynamoDBLogStoreSuite?

also.. why even have this class as a placeholder?

File: build.sbt (hunk 2 above)
Comment (scottsand-db) [reply build.sbt:184]: We're not ready to release this yet

File: storage/src/main/java/io/delta/storage/external/BaseExternalLogStore.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+import io.delta.storage.HadoopFileSystemLogStore;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+public abstract class BaseExternalLogStore extends HadoopFileSystemLogStore {
```
[the remark this answers is not in the document]
Comment (scottsand-db) [reply storage/src/main/java/io/delta/storage/external/BaseExternalLogStore.java:27]: Yup will be added later :) This is just a skeleton

File: build.sbt (hunk 1 above)
Comment (scottsand-db) [reply build.sbt:187]: cc @mrk-its

Comment (mrk-its) [reply build.sbt:187]: 1.7.4 is the version we were using from the beginning of DynamoDBLogStore. AFAIR ir was version used by hadoop-aws 2.7 / spark 2.4 (at least on AWS EMR). And it still works with latest EMR versions, so we didn't change it. But we can of course try to bump it (I even tried to do this update some time ago, but immediately ran into sam problems with writing to S3, so it needs to be done very carefully)

File: build.sbt [hunk 5]
```diff
@@ -34,6 +34,7 @@ lazy val commonSettings = Seq(
 
 lazy val core = (project in file("core"))
   .dependsOn(storage)
+  .dependsOn(storageDynamodb % "test->compile") // delta-core tests depend on storageDynamodb
```
Comment (zsxwing) [inline build.sbt:37]: Why not make `storageDynamodb`'s `test` depends on `core`'s `test`?

File: storage/src/main/java/io/delta/storage/external/BaseExternalLogStore.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (zsxwing) [reply storage/src/main/java/io/delta/storage/external/BaseExternalLogStore.java:27]: Why do we add this into storage?

File: core/src/test/scala/org/apache/spark/sql/delta/ExternalLogStoreSuite.scala (hunk 3 above)
Comment (zsxwing) [reply core/src/test/scala/org/apache/spark/sql/delta/ExternalLogStoreSuite.scala:20]: It's better to not rush to build a public `ExternalLogStore` API in the first version. We can keep it in delta-storage-dynamodb for now.

File: storage/src/main/java/io/delta/storage/external/BaseExternalLogStore.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (scottsand-db) [reply storage/src/main/java/io/delta/storage/external/BaseExternalLogStore.java:27]: External means we are depending on some other, external storage system (e.g. DynamoDB, Zookeeper) to provide the mutual exclusion that S3 is missing.

File: build.sbt (hunk 5 above)
Comment (scottsand-db) [reply build.sbt]: ACK

**Pull request closed without merging**